### ISSUE TITLE
Low level connection exceptions are terminal when they could be handled like other connection issues and retried

### ISSUE THREAD

**Issue by scottf** [nats-io/nats.java#974]
### Observed behavior

Low level connection issues at the socket layer that currently throw terminal exceptions and stop the ability to retry connections.

### Expected behavior

The exception is handled, the connection try is marked as a failure and retry logic continues.

### Server and client version

Latest client / server N/A

### Host environment

N/A

### Steps to reproduce

TBDoc'ed

**Comment by scottf** [comment]
Against 2.16.14
```
java.io.IOException: java.net.ConnectException: Connection refused (Connection refused)
    at io.nats.client.impl.SocketDataPort.connect(SocketDataPort.java:65)
    at io.nats.client.impl.NatsConnection.tryToConnect(NatsConnection.java:389)
    at io.nats.client.impl.NatsConnection.connect(NatsConnection.java:178)
    at io.nats.client.impl.NatsImpl.createConnection(NatsImpl.java:29)
    at io.nats.client.Nats.createConnection(Nats.java:287)
    at io.nats.client.Nats.connect(Nats.java:193)
    at com.mastercard.eventframework.nats.utils.ConnectionHelper.getConnection(ConnectionHelper.java:92)
```

**Comment by scottf** [comment]
Turns out this is an effect of not allowing reconnect on connect. I have added easier ways to make the initial connection in the PR and updated the readme

**Comment by scottf** [comment]
Unit tests and additional api specifically for reconnect on connect in this PR https://github.com/nats-io/nats.java/pull/1019

**Comment by scottf** [comment]
Addressed by #1019
